modules/articles/client/services/articles.client.service.js

```diff
@@ -8,12 +8,29 @@
   ArticlesService.$inject = ['$resource'];
 
   function ArticlesService($resource) {
-    return $resource('api/articles/:articleId', {
+    var Article = $resource('api/articles/:articleId', {
       articleId: '@_id'
     }, {
       update: {
         method: 'PUT'
       }
     });
+
+    angular.extend(Article.prototype, {
+      createOrUpdate: function () {
+        var article = this;
+        return createOrUpdate(article);
+      }
+    });
+
+    return Article;
+
+    function createOrUpdate(article) {
```

Comment:
I think having the function declarations after the return statement make it easier to read, and understand the implementation. 

It follows this [style from John Papa's guide](https://github.com/johnpapa/angular-styleguide/blob/master/a1/README.md#accessible-members-up-top). I don't feel super strong about it, but I think it would help others that are following, or aware of, the style-guide.